### DISCUSSION TITLE
Interactive Keyboard dismissing

### DISCUSSION THREAD

**Discussion by kirillzyusko** [kirillzyusko/react-native-keyboard-controller#6] [Ideas]
It would be good to be able to close keyboard in interactive mode. 

iOS already supports it via `keyboardDismissMode="interactive"` prop on `ScrollView` (though current implementation of this library is not updating animated values returned from the hook). On Android it can be implemented through `WindowInsetsAnimationController`.

Something similar was already implemented on iOS: https://github.com/wix/react-native-ui-lib/tree/master/lib/ios/reactnativeuilib/keyboardtrackingview

**Comment by arnaud9145** [comment]
Hello @kirillzyusko, have you made any progress on that subject ?

I'm looking for a way to do an interactive keyboard dismiss for a chat app, and your library looks appealing, but useKeyboardHandler does not follow the interactive dismiss on ios for now.

**Reply by kirillzyusko** [reply]
Hi @arnaud9145 
I'm currently working on it and experimenting with the API.
You can check the progress here: https://github.com/kirillzyusko/react-native-keyboard-controller/compare/main...feature/interactive-keyboard-android
Android implementation is still buggy and not ready for production (it has flicks/jumps and other unpleasant UI things). iOS implementation is temporarily postponed, because I want to roll out Android first and stabilize API (after that I'll add iOS implementation).

**Comment by kirillzyusko** [comment]
This feature was added in `1.5.0` 😊
